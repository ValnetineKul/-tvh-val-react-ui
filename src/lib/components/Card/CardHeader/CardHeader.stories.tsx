import React from 'react';
import { createMeta } from '../../../story-utils';
import CardHeader from './CardHeader';

const Template = (args: React.ComponentProps<typeof CardHeader>) => <CardHeader {...args} />;

export const Primitive = Template.bind({});

Primitive.args = {
  title: 'Card Header',
  subtitle: 'Subtitle',
};

Primitive.storyName = 'CardHeader';

export default createMeta({
  component: CardHeader,
  title: 'Components/Cards/CardHeader',
});
