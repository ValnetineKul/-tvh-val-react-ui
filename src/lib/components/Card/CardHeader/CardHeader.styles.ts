import { makeStyles } from '../../../../themes/core';

const useStyles = makeStyles({ name: 'CardHeader' })((theme) => ({
  root: {
    padding: theme.layoutSpacing['spacing/400'],
    color: theme.color['text/onSurfaceCurrent/default'],
  },
}));

export default useStyles;
