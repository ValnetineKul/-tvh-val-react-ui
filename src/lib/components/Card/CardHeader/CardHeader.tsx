import React from 'react';
import type { FC } from 'react';
import classNames from 'classnames';

import Typography from '../../Typography';
import useStyles from './CardHeader.styles';

export interface CardHeaderProps {
  title: string;
  subtitle?: string;
  className?: string;
}

const CardHeader: FC<CardHeaderProps> = ({ title, subtitle, className }) => {
  const { classes } = useStyles();
  return (
    <div className={classNames(classes.root, className)}>
      <Typography variant="h3">{title}</Typography>
      <Typography component="span" variant="h4" secondary>
        {subtitle}
      </Typography>
    </div>
  );
};

export default CardHeader;
