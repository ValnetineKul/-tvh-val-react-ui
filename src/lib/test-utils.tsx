import userEvent from '@testing-library/user-event';
import { fireEvent, render } from '@testing-library/react';
import { renderHook } from '@testing-library/react-hooks';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { theme } from '../themes/tvh';
import { createTheme, ThemeProvider, SurfaceProvider } from '../themes/core';

export const muiTheme = createTheme(theme);

const AllTheProviders: React.FC = ({ children }) => {
  return (
    <MemoryRouter>
      <ThemeProvider theme={muiTheme}>
        <SurfaceProvider color="100">{children}</SurfaceProvider>
      </ThemeProvider>
    </MemoryRouter>
  );
};

const customRender = (component: React.ReactElement, options?: Parameters<typeof render>[1]) =>
  render(component, { wrapper: AllTheProviders, ...options });

const customRenderHook: typeof renderHook = (hook, options) =>
  renderHook(hook, { wrapper: AllTheProviders, ...options });

export const initResponsiveTest = (size: number | 'Mobile' | 'Tablet' | 'Desktop' | 'LgDesktop' | 'XlDesktop') => {
  let width: number;
  if (size === 'Mobile') width = muiTheme.breakpoints.values.sm - 1;
  else if (size === 'Tablet') width = muiTheme.breakpoints.values.sm;
  else if (size === 'Desktop') width = muiTheme.breakpoints.values.md;
  else if (size === 'LgDesktop') width = muiTheme.breakpoints.values.lg;
  else if (size === 'XlDesktop') width = muiTheme.breakpoints.values.xl;
  else width = size;

  Object.defineProperty(window, 'innerWidth', {
    writable: true,
    configurable: true,
    value: width,
  });

  window.matchMedia = jest.fn().mockImplementation((query) => {
    const queryWidthInterval: string[] = query.match(/\d+(\.\d+)?/g);

    let matches;
    if (queryWidthInterval.length === 2) {
      matches = width >= +queryWidthInterval[0] && width < +queryWidthInterval[1];
    } else {
      const queryWidth = +queryWidthInterval[0];
      const isDown = /max/.test(query);

      matches = isDown ? width <= queryWidth : width >= queryWidth;
    }

    return {
      matches,
      media: query,
      onchange: null,
      addListener: jest.fn(),
      removeListener: jest.fn(),
    };
  });
};

const customUserEvent = Object.assign(userEvent, {
  mouseDown: fireEvent.mouseDown,
  mouseUp: fireEvent.mouseUp,
  mouseMove: fireEvent.mouseMove,
  scroll: fireEvent.scroll,
  error: fireEvent.error,
});

export { screen, within, waitFor, act, waitForElementToBeRemoved } from '@testing-library/react';
// override render method
export { customRender as render, customRenderHook as renderHook, customUserEvent as userEvent };
