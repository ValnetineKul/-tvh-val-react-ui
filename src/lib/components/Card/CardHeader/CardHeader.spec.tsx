import React from 'react';
import { render } from '../../../test-utils';
import CardHeader from './CardHeader';

describe('CardHeader', () => {
  it('Should render correctly', () => {
    const { container } = render(<CardHeader title="Card title" subtitle="Card subtitle" />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
