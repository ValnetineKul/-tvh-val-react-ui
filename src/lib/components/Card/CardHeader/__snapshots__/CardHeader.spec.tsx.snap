// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardHeader Should render correctly 1`] = `
<div
  class="tss-15t76xy-CardHeader-root"
>
  <h3
    class="tss-0-Typography-headerFunctional-ref tss-11gt335-Typography-root-Typography-headerFunctional-Typography-h3"
  >
    Card title
  </h3>
  <span
    class="tss-0-Typography-headerFunctional-ref tss-29ec5r-Typography-root-Typography-headerFunctional-Typography-h4-Typography-secondary"
  >
    Card subtitle
  </span>
</div>
`;
